notification-api.md

```diff
@@ -0,0 +1,297 @@
+         read-out preempted it.
+
+### Failed or delayed actions
+
+According to common screen reader etiquette, user actions are assumed to be successful by
+virtue of issuing the command to do the action itself (no specific confirmation of the action
+needed); however, if the action fails or is delayed, the user should then be notified. In these
+unexpected cases users should be notified, otherwise their understanding about the state of the
+app will be off.
+
+3. **Longer than usual.** User completes typing a mail message, presses _send_. In the normal
+    flow, no confirmation of "sent" is needed because this is assumed by the action, and focus
+    is redirected to the message list or next message. However, due to some networking conditions,
+    the _send_ action is taking longer than usual. The user should hear "message is taking longer
+    than usual to send".
+
+    3.1. **High overall priority.** After pressing _send_, the user resumes navigating through
+          the message list. At the conclusion of reading an existing email subject line, the AT
+          breaks in with "message is taking longer than usual to send". (Note, there should be
+          some means of separating this announcement from the prior email text, lest it be
+          considered a part of the email subject line).
+
+4. **Fail to paste.** User thought they had copied some text onto the clipboard, but in the
+    context of editing when they issue the paste keyboard shortcut, nothing is in the clipboard
+    and nothing pastes into the app. In this case, it is appropriate for the app to note the
+    failed action: "failed to paste, clipboard empty". 
+
+### Secondary actions
+
+In addition to a primary (implicit) action, some actions have secondary or follow-up effects 
+that should be announced beyond the immediate effect of the primary action.
+
+5. **Auto fill.** In a spreadsheet, an action that sets a cell's value may be assumed to happen
+    (no announcement) or could be announced as a side-effect of changing the cell's value (e.g., 
+    using a live region). In either case this would be the normal expectation for the user. However,
+    as a result of setting the value, the spreadsheet triggered a secondary action of autofilling a
+    range of corresponding values in the cell's column. This is an opportunity for the app to 
+    additionally announce "autofilled values in cells A2 through A20".
+
+## Proposed Solution
+
+(This version inspired by [UIA Notification API](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcoreapi/nf-uiautomationcoreapi-uiaraisenotificationevent))
+
+For use cases that don't have an express UI tie-in, it makes sense to provide a solution
+not expressly tied to HTML element, but for which other document-centric information can
+be inferred (such as language).
+
+```js
+// As the assistive technology to notify the user, given a specific string
+document.ariaNotify( "Selected text is now glowing blue." );
```

Comment:
I feel like it'd be better to allow this on any Element, not the document,
because ideally it should still scope the announcement to the part of
the document that it applies to. In all of these examples there's a clear
associated element or container.

OK to just merge this as-is and we can discuss this in a follow-up PR
or a bug.

Reply:
Sounds good. I'll follow-up with an issue for the above.